tests/pipeline/test_pipeline.py

```diff
@@ -632,3 +636,123 @@ def test_validate_topic_and_component_same_name(self):
         edges = list(pipeline.components.graph.edges)
         assert component == f"component-{topic}"
         assert (component, topic) in edges
+
+    @pytest.mark.asyncio()
+    async def test_parallel_execution_graph(self):
+        pipeline = kpops.generate(
+            RESOURCE_PATH / "parallel-pipeline/pipeline.yaml",
+            pipeline_base_dir=PIPELINE_BASE_DIR_PATH,
+            defaults=RESOURCE_PATH / "parallel-pipeline",
+            config=RESOURCE_PATH / "parallel-pipeline/config.yaml",
+        )
+
+        called_component = AsyncMock()
+
+        sleep_table_components = {
+            "transaction-avro-producer-1": 1,
+            "transaction-avro-producer-2": 0,
+            "transaction-avro-producer-3": 2,
+            "transaction-joiner": 3,
+            "fraud-detector": 2,
+            "account-linker": 0,
+            "s3-connector-1": 2,
+            "s3-connector-2": 1,
+            "s3-connector-3": 0,
+        }
+
+        async def name_runner(component: PipelineComponent):
+            await asyncio.sleep(sleep_table_components[component.name])
+            await called_component(component.name)
+
+        execution_graph = pipeline.build_execution_graph_from(
+            list(pipeline.components), False, name_runner
+        )
+
+        await execution_graph
+
+        assert called_component.mock_calls == [
+            mock.call("transaction-avro-producer-2"),
+            mock.call("transaction-avro-producer-1"),
```

Comment:
Why is the `transaction-avro-producer-1` called after the `transaction-avro-producer-2`?

Reply:
because the sleep is 0. This is actually what I wanted to test, that this 3 components can be executed in parallel.